### PR TITLE
Skip adding hook by rake if one already added

### PR DESCRIPTION
**Pull request by newmen**
Now we can run rake task without care about webhook duplication

### CODE REVIEW

File: lib/tasks/gitlab/web_hook.rake [hunk 1]
```diff
@@ -7,15 +7,24 @@ namespace :gitlab do
 
       projects = find_projects(namespace_path)
 
+      # Select projects which already have setting hook
+      ns_project_ids = projects.select(:id).reorder('').map(&:id)
+      project_ids_with_hook = ProjectHook.where(project_id: ns_project_ids, url: web_hook_url)
```
Comment (jvanbaarsen) [inline lib/tasks/gitlab/web_hook.rake]: Would it be possible to do:
```
ProjectHook.where(project_id: ns_project_ids, url: web_hook_url).pluck(:project_id)
```

File: lib/tasks/gitlab/web_hook.rake [hunk 2]
```diff
@@ -7,15 +7,23 @@ namespace :gitlab do
 
       projects = find_projects(namespace_path)
 
+      with_hook_scope = ProjectHook.where(url: web_hook_url)
+      with_hook_scope = with_hook_scope.where(project: projects) unless projects == Project # there is Rails bug
```
Comment (jvanbaarsen) [inline lib/tasks/gitlab/web_hook.rake]: Can you clarify what rails bug? Is an issue report already present?

Comment (newmen) [inline lib/tasks/gitlab/web_hook.rake]: The problem is:
```
Loading production environment (Rails 4.1.9)
irb(main):001:0> ProjectHook.where(url: 'asdf', project: Project).to_sql
=> "SELECT \"web_hooks\".* FROM \"web_hooks\"  WHERE \"web_hooks\".\"type\" IN ('ProjectHook') AND \"web_hooks\".\"url\" = 'asdf' AND \"web_hooks\".\"project_id\" = 0  ORDER BY \"web_hooks\".\"created_at\" DESC, \"web_hooks\".\"id\" DESC"
```
So, when the `Project` isn't scoped then request contains incorrect condition.
But when we're use the scoped `Project` like:
```
irb(main):006:0> ProjectHook.where(url: 'asdf', project: Project.all.reorder('')).to_sql
=> "SELECT \"web_hooks\".* FROM \"web_hooks\"  WHERE \"web_hooks\".\"type\" IN ('ProjectHook') AND \"web_hooks\".\"url\" = 'asdf' AND \"web_hooks\".\"project_id\" IN (SELECT \"projects\".\"id\" FROM \"projects\")  ORDER BY \"web_hooks\".\"created_at\" DESC, \"web_hooks\".\"id\" DESC"
```
then resulting request is pretty correct.

I can change the 11 line of my patch or can change the method which returns the unscoped `Project` for `Project.all.reorder('')`. How to do better?

P.S. I don't know is it bug or not. But I think it is non-obvious behavior.

**Comment by jvanbaarsen** [comment]
This merge request has been closed because a request for more information has not been reacted to for more than 2 weeks. If you respond and conform to the merge request guidelines in our [contributing guidelines](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CONTRIBUTING.md#pull-requests) we will reopen this merge request.

**Pull request closed without merging**